File: Exercicio ordenaçao java/ordenacao.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ordenacao {
    public static List<Integer> quickSort(List<Integer> array) {
        if (array.size() <= 1) {
            return array;
        }


        int pivo = array.get(0);

        List<Integer> menores = new ArrayList<>();
        List<Integer> maiores = new ArrayList<>();

        for (int i = 1; i < array.size(); i++) {
            if (array.get(i) < pivo) {
                menores.add(array.get(i));
            } else {
                maiores.add(array.get(i));
            }
        }

        List<Integer> resultado = new ArrayList<>(quickSort(menores));
        resultado.add(pivo);
        resultado.addAll(quickSort(maiores));

        return resultado;
    }

    public static void main(String[] args) {

        List<Integer> numeros = Arrays.asList(8, 3, 1, 7, 0, 10, 2);

        List<Integer> numerosOrdenados = quickSort(numeros);

        System.out.println("Lista ordenada: " + numerosOrdenados);
    }
}

File: Exercicio ordenação/ordenacao.js
function quickSort(array) {

    if (array.length <= 1) {
        return array;
    }

    let pivo = array[0];

    let menores = [];
    let maiores = [];

    for (let i = 1; i < array.length; i++) {
        if (array[i] < pivo) {
            menores.push(array[i]);
        } else {
            maiores.push(array[i]); 
        }
    }

    return [...quickSort(menores), pivo, ...quickSort(maiores)];
}

let numeros = [8, 3, 1, 7, 0, 10, 2];
let numerosOrdenados = quickSort(numeros);
console.log("Lista ordenada:", numerosOrdenados);